pkg/cmd/pulumi/operations/io.go

```diff
@@ -104,3 +109,30 @@ func getRefreshOption(proj *workspace.Project, refresh string) (bool, error) {
 	// the default functionality right now is to always skip a refresh
 	return false, nil
 }
+
+// configureCopilotOptions configures display options related to Copilot features based on the command line flags and
+// environment variables.
+func configureCopilotOptions(copilotEnabledFlag bool, cmd *cobra.Command, displayOpts *display.Options,
+	isDIYBackend bool,
+) {
+	// Handle copilot-summary flag and environment variable If flag is explicitly set (via command line), use that value
+	// Otherwise fall back to environment variable, then default to false
+	var showCopilotFeatures bool
+	if cmd.Flags().Changed("copilot") {
+		showCopilotFeatures = copilotEnabledFlag
+	} else {
+		showCopilotFeatures = env.CopilotEnabled.Value()
+	}
+	logging.V(7).Infof("copilot flag=%v, PULUMI_COPILOT=%v, using value=%v",
+		copilotEnabledFlag, env.CopilotEnabled.Value(), showCopilotFeatures)
+
+	// Do not enable any copilot features if we are using a DIY backend
+	if isDIYBackend {
+		logging.Warningf("Copilot features are not available with DIY backends.")
```

Comment:
Wait, are we always warning, regardless of the value of the options?

Reply:
Ooops! Great catch, shifted code around a bit too quickly.